### PR TITLE
docs: Add types in injection token examples

### LINKED ISSUE #20759: docs-bug(dialog): Suggest using strong types for MAT_DIALOG_DATA
#### Documentation Feedback

In the **Sharing data with Dialog component** section, the type of `MAT_DIALOG_DATA` in the described example is set to `any`. Given that it is encouraged to use strongly typing in Angular projects, I would suggest to use the following interface for the example:

```
interface DialogData {
 name: string;
}

export class YourDialog {
  constructor(@Inject(MAT_DIALOG_DATA) public data: DialogData) { }
}
```

**Affected documentation page:** https://material.angular.io/components/dialog/overview#sharing-data-with-the-dialog-component

I would love to work on a related PR if you find it useful. 🙏

### PR DESCRIPTION
**Pull request by bampakoa**
Use strong typing for the MAT_DIALOG_DATA, MAT_BOTTOM_SHEET_DATA and MAT_SNACK_BAR_DATA injection token examples. Closes #20759

### CODE REVIEW

File: src/material/bottom-sheet/bottom-sheet.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ import {MAT_BOTTOM_SHEET_DATA} from '@angular/material/bottom-sheet';
   template: 'passed in {{ data.names }}',
 })
 export class HobbitSheet {
-  constructor(@Inject(MAT_BOTTOM_SHEET_DATA) public data: any) { }
+  constructor(@Inject(MAT_BOTTOM_SHEET_DATA) public data: { names: string[] }) { }
```
Comment (crisbeto) [inline src/material/bottom-sheet/bottom-sheet.md:48]: nit: usually we don't have spaces inside of object literals (e.g. `{names: string[]}`).

**Review by crisbeto — APPROVED** [review]
LGTM

File: src/material/bottom-sheet/bottom-sheet.md (hunk 1 above)
Comment (bampakoa) [reply src/material/bottom-sheet/bottom-sheet.md:48]: @crisbeto I am curious, is this a coding style that you follow?

Comment (crisbeto) [reply src/material/bottom-sheet/bottom-sheet.md]: It's how we've always been writing them. It might be coming from the Google JS style guide.

**Review by jelbourn — APPROVED** [review]
LGTM

**Pull request merged**